addons/account/models/account_move.py

```diff
@@ -2295,6 +2319,15 @@ def copy(self, default=None):
 
         return copied_am
 
+    @api.model
+    def default_get(self, default_fields):
+        values = super().default_get(default_fields)
+        if values.get('journal_id') and values.get('move_type'):
+            journal = self.env['account.journal'].browse(values['journal_id'])
+            if self._use_quick_edit_mode(journal.company_id, journal):
+                values['invoice_date'] = self._get_quick_edit_suggestions(0, True, values.get('move_type'), journal.id, None, None).get('invoice_date', False)
+        return values
```

Comment:
Let's wait on https://github.com/odoo/odoo/pull/96134 to be able to write this without needing `_use_quick_edit_mode`
I also rewrote `_get_quick_edit_suggestions` in that branch